godtools/Managers/LanguagesManager.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  LanguagesManager.swift
+//  godtools
+//
+//  Created by Ryan Carlson on 4/18/17.
+//  Copyright © 2017 Cru. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import Alamofire
+import PromiseKit
+import Spine
+import MagicalRecord
+
+class LanguagesManager: NSObject {
+    static let shared = LanguagesManager()
+    
+    let path = "/languages"
+    let serializer = Serializer()
+    
+    var languages = [Language]()
+    
+    override init() {
+        super.init()
+        serializer.registerResource(LanguageResource.self)
+    }
+    
+    func loadFromDisk() -> Promise<[Language]> {
+        languages = Language.mr_findAll() as! [Language]
+        
+        for language in languages {
+            language.localizedName = NSLocale.current.localizedString(forLanguageCode: language.code!)
+        }
+        
+        languages = languages.sorted { (language1, language2) -> Bool in
+            return language1.localizedName!.compare(language2.localizedName!).rawValue < 0
+        }
+        
+        return Promise(value:languages)
+    }
+    
+    func loadFromRemote() -> Promise<[Language]> {
+        UIApplication.shared.isNetworkActivityIndicatorVisible = true
+        
+        return Alamofire.request(URL(string: "\(GodToolsConstants.kApiBase)/\(path)")!).responseData().then { data -> Promise<[Language]> in
+            let jsonDocument = try! self.serializer.deserializeData(data)
+            
+            if jsonDocument.data != nil {
+                MagicalRecord.save(blockAndWait: { (context) in
+                    for element in jsonDocument.data! {
+                        let languageResource = element as! LanguageResource;
+                        let language = Language.mr_findFirstOrCreate(byAttribute: "remoteId", withValue: languageResource.id!, in: context)
+                        language.code = languageResource.code
+                    }
+                })
+            }
+            
+            return self.loadFromDisk()
+        }
+    }
+}
+
+extension LanguagesManager: UITableViewDelegate {
+    
+}
+
+extension LanguagesManager: UITableViewDataSource {
```

Comment:
I like the extensions, could you please set them in separated files? So in the future we will have a group for the extensions for the datasources and delegates, separated by each resource